app/modules/academics/router.py

```
@@ -1,80 +1,80 @@
+    return services.StreamService(db).create_stream(principal.school_id, data)
+@router.patch("/streams/{stream_id}", response_model=schemas.StreamResponse)
+def update_stream(stream_id: int, data: schemas.StreamUpdate, principal: Principal = Depends(require_role("admin")), db: Session = Depends(get_db)):
+    return services.StreamService(db).update_stream(principal.school_id, stream_id, data)
+
+@router.get("/streams/{stream_id}/students", response_model=List[schemas.StreamStudentResponse])
+def list_stream_students(stream_id: int, principal: Principal = Depends(require_role("viewer", "teacher", "admin")), db: Session = Depends(get_db)):
+    stream = services.StreamService(db).get_stream_by_id(principal.school_id, stream_id)
+    return db.query(student_models.Student).filter(student_models.Student.school_id == principal.school_id, student_models.Student.stream_id == stream.id).order_by(student_models.Student.last_name, student_models.Student.first_name).all()
+
+@router.post("/streams/{stream_id}/students", response_model=schemas.StreamStudentResponse)
+def assign_student_to_stream(stream_id: int, data: schemas.StreamAssignment, principal: Principal = Depends(require_role("admin", "scheduler")), db: Session = Depends(get_db)):
```

Ruff (0.16.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)